libs/api/domains/national-registry-v3/src/lib/nationalRegistryV3.service.ts

```diff
@@ -0,0 +1,310 @@
+                ?.map((l) => l.kennitala)
+                .includes(data.kennitala),
+            }
+          })
+          .filter((Boolean as unknown) as ExcludesFalse),
+      )) ?? []
+
+    return custodianData.filter(
+      (custodian): custodian is Custodian => custodian != null,
+    )
+  }
+
+  async getChildGuardianship(
+    user: User,
+    childNationalId: string,
+  ): Promise<ChildGuardianship | null> {
+    const data = await this.fetchData(user.nationalId)
+
+    if (!data.forsja?.born) {
+      return null
+    }
+
+    const childrenNationalIds = data.forsja.born.map((b) => b.barnKennitala)
+
+    const isChildOfUser = childrenNationalIds.some(
+      (childId) => childId === childNationalId,
+    )
+
+    if (!isChildOfUser) {
+      return null
+    }
+
+    const childData = await this.fetchData(childNationalId)
+
+    const legalDomicileParent = childData.logforeldrar
+      ? childData.logforeldrar?.logForeldrar
+          ?.map((l) => l.logForeldriKennitala)
+          .filter((Boolean as unknown) as ExcludesFalse)
+      : []
+
+    return {
+      childNationalId,
+      legalDomicileParent,
+    }
+  }
+
+  async getSpouse(nationalId: string): Promise<Spouse | null> {
+    const data = await this.fetchData(nationalId)
+
+    return (
+      (data.hjuskaparstada && {
+        nationalId: data.hjuskaparstada.makiKennitala,
+        fullName: data.hjuskaparstada.makiNafn,
+        maritalStatus: data.hjuskaparstada.hjuskaparstadaTexti,
+        cohabitation: data.hjuskaparstada.sambudTexti,
+      }) ??
+      null
+    )
+  }
+
+  async getDomicilePopulace(
+    nationalId: string,
+  ): Promise<DomicilePopulace | null> {
+    const data = await this.fetchData(nationalId)
+
+    return (
+      (data.logheimilistengsl &&
+        ({
+          legalDomicileId: data.logheimilistengsl.logheimilistengsl,
+          populace: data.logheimilistengsl.logheimilismedlimir?.map(
+            (person) => ({
+              nationalId: person.kennitala,
+              name: person.nafn,
+            }),
+          ),
+        } as DomicilePopulace)) ??
+      null
+    )
+  }
+
+  async getCitizenship(nationalId: string): Promise<Citizenship | null> {
+    const data = await this.fetchData(nationalId)
+
+    return (
+      (data.rikisfang && {
+        name: data.rikisfang.rikisfangLand,
+        code: data.rikisfang.rikisfangKodi,
+      }) ??
+      null
+    )
+  }
+
+  async getChildren(nationalId: string): Promise<Array<Person> | null> {
+    const parentData = await this.fetchData(nationalId)
+
+    const children = parentData.forsja?.born?.length
+      ? parentData.forsja.born
+      : []
+
+    const childDetails: Array<Person | null> = await Promise.all(
+      children.map(async (child) => {
+        if (!child.barnKennitala || !child.barnNafn) {
+          return null
+        }
+
+        const childDetails = await this.fetchData(child.barnKennitala)
```

Comment:
I wonder if we should use the new batch endpoint for querying children and parents?

Reply:
I'm unfamiliar, can you elaborate?